packages/platform/atoms/cal-provider/BaseCalProvider.tsx

```diff
@@ -13,14 +17,26 @@ import http from "../lib/http";
 import { Toaster } from "../src/components/ui/toaster";
 import type { CalProviderProps } from "./CalProvider";
 
-type translationKeys = keyof typeof enTranslations;
+type enTranslationKeys = keyof typeof enTranslations;
+type frTranslationKeys = keyof typeof frTranslations;
+type deTranslationKeys = keyof typeof deTranslations;
+type esTranslationKeys = keyof typeof esTranslations;
+type ptBrTranslationKeys = keyof typeof ptBrTranslations;
+type translationKeys =
```

Comment:
Here in BaseCalProvider translation keys is union of all language files keys, but in In CalProvider we have translationKeys as `type translationKeys = keyof typeof enTranslations;`, which are only english keys, and also they are not having the logic of a discriminative union in CalProvider:

```
export type CalProviderProps = {
  labels?: Record<translationKeys, string>;
  language?: "en" | "fr" | "pt-BR" | "de" | "es";
```

aka we can pass any "language", like "fr" and then a key in "labels" that only exists in english. So I would:
1. Also here make `type translationKeys = keyof typeof enTranslations;`
2. In `getTranslation` if "de" or other language returns undefined, then fallback to english:

```
case "de":
      return deTranslations[key] || enTranslations[key];
```

and then promise to work on having all translations in place for all supported languages and roll with an assumption that whatever key is in english is for other languages too, and if someone reports that it shows up in english then we know we have a missing key.

Reply:
I am writing this because I also see in `useAtomsContext.ts` having `type translationKeys = keyof typeof enTranslations;`, so we have defined `type translationKeys` in 3 places:
1. CalProvider
2. BaseCalProvider
3. useAtomsContext

We should define it in 1 place and re-use the same type everywhere.

Reply:
adding discriminative union so that labels and language always match